homeassistant/components/myuplink/config_flow.py

```diff
@@ -23,3 +28,30 @@ def logger(self) -> logging.Logger:
     def extra_authorize_data(self) -> dict[str, Any]:
         """Extra data that needs to be appended to the authorize url."""
         return {"scope": " ".join(OAUTH2_SCOPES)}
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.config_entry_reauth = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+            )
+
+        return await self.async_step_user()
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create or update the config entry."""
+        if self.config_entry_reauth:
+            return self.async_update_reload_and_abort(
```

Comment:
Everything works fine when running hass, the entry is updated, the integration is reloaded and the flow is terminated. 
However, when running test_config_flow I can see that self.config_entry_reauth is None here. Therefore the abort-helper is never run and the coverage lacks this one single statement.

Reply:
The current reauth test will fail when the WRITESYSTEM scope is in place in the mocked entry. I was too tired to learn how to make an elegant fixture for that entry so i just duplicated a lot of code.

The snapshot is fixed.